tools/dependency-check/check-dependencies.sh

```diff
@@ -34,14 +34,14 @@ self_modules_txt='tools/dependency-check/self-modules.txt'
 # store all third part dependencies
 third_party_dependencies_txt='tools/dependency-check/third-party-dependencies.txt'
 
-mkdir $decompress_conf || true
+mkdir -p $decompress_conf
 tar -zxf build/eventmesh*.tar.gz -C $decompress_conf
 
 ./gradlew printProjects | grep '.jar' > "$self_modules_txt"
 
-find "$decompress_conf" -name "*.jar" -exec basename {} \; | uniq | sort > "$all_dependencies_txt"
+find "$decompress_conf" -name "*.jar" -exec basename {} \; | sort | uniq > "$all_dependencies_txt"
 
-grep -wvf "$self_modules_txt" "$all_dependencies_txt" | uniq | sort > "$third_party_dependencies_txt"
+grep -wvf "$self_modules_txt" "$all_dependencies_txt" | sort | uniq > "$third_party_dependencies_txt"
 
 # If the check is success it will return 0
```

Comment:
It seems switching `sort` and `uniq` won't change the result, but speed up a little, right?

Reply:
These are old POSIX commands that not necessarily do what you would expect. E.g. what `uniq` does is:

> Filter **adjacent** matching lines from INPUT (or standard input), writing to OUTPUT (or standard output).

(cf. [man uniq](https://man7.org/linux/man-pages/man1/uniq.1.html), emphasis mine).

That is why you need to sort the file before running `uniq` or you'll end up with duplicate entries.